src/Plug.vala

```diff
@@ -36,15 +33,34 @@ public class About.Plug : Switchboard.Plug {
 
     public override Gtk.Widget get_widget () {
         if (main_grid == null) {
-            setup_info ();
-            setup_ui ();
+            main_grid = new Gtk.Grid ();
+
+            stack = new Gtk.Stack ();
+
+            operating_system_view = new OperatingSystemView ();
+            stack.add_titled (operating_system_view, "os", _("Operating System"));
+
+            var hardware_view = new HardwareView ();
+            stack.add_titled (hardware_view, "hardware", _("Hardware"));
+
+            var firmware_view = new FirmwareView ();
+            stack.add_titled (firmware_view, "firmware", _("Firmware"));
+
+            var stack_switcher = new Gtk.StackSwitcher ();
+            stack_switcher.stack = stack;
+            stack_switcher.halign = Gtk.Align.CENTER;
+            stack_switcher.homogeneous = true;
+            stack_switcher.margin = 24;
+
+            main_grid.attach (stack_switcher, 0, 0, 1, 1);
+            main_grid.attach (stack, 0, 1, 1, 1);
```

Comment:
attach uses a width and height of 1 by default so this can be:
```suggestion
            main_grid.attach (stack_switcher, 0, 0);
            main_grid.attach (stack, 0, 1);
```